pkg/server/request.go

```diff
@@ -83,28 +83,31 @@ func buildFileRequest(url string, fieldData map[string]string, fileFieldData map
 //
 // Returns:
 //   - error: An error if any step of the file processing fails. Nil if the process is successful.
-func ProcessFileRequest(endpoint string, uploadOptions map[string]string, fileFieldData map[string]string, timeout int, retries int, fileName string, dryRun bool) error {
+func ProcessFileRequest(endpoint string, uploadOptions map[string]string, fileFieldData map[string]string, timeout int, retries int, fileName string, dryRun bool, logger log.Logger) error {
 	req, err := buildFileRequest(endpoint, uploadOptions, fileFieldData)
 	if err != nil {
 		return fmt.Errorf("error building file request: %w", err)
 	}
 
 	if !dryRun {
-		log.Info(fmt.Sprintf("Uploading %s to %s", filepath.Base(fileName), endpoint))
+		logger.Info(fmt.Sprintf("Uploading %s to %s", filepath.Base(fileName), endpoint))
 
-		err = processRequest(req, timeout, retries)
+		err = processRequest(req, timeout, retries, logger)
 
 		if err != nil {
 			if strings.Contains(err.Error(), "409") {
-				log.Warn(fmt.Sprintf("Duplicate file detected, skipping upload of %s", filepath.Base(fileName)))
+				logger.Warn(fmt.Sprintf("Duplicate file detected, skipping upload of %s", filepath.Base(fileName)))
 			} else {
 				return err
 			}
 		} else {
-			log.Success(fmt.Sprintf("Uploaded %s", filepath.Base(fileName)))
+			logger.Info("Uploaded " + filepath.Base(fileName))
 		}
 	} else {
-		log.Info(fmt.Sprintf("(dryrun) Skipping upload of %s to %s", filepath.Base(fileName), endpoint))
+		logger.Info(fmt.Sprintf("(dryrun) Skipping upload of %s to %s", filepath.Base(fileName), endpoint))
+		logger.Info("(dryrun) Upload payload:")
+		prettyUploadOptions, _ := utils.PrettyPrintMap(uploadOptions)
```

Comment:
> When the --verbose and --dry-run flag are passed, the CLI prints out the payload to be sent to the API, minus the file data.

Does this correctly omit the file data as described?

Reply:
Yes, `uploadOptions` doesn't contain the file data, but does include the name of the file that is being uploaded. The file data is stored within `fileFieldData`